libs/@blockprotocol/type-system/rust/src/schema/data_type/constraint/array.rs

```diff
@@ -43,7 +46,27 @@ pub enum ArrayItemConstraints {
     String(StringSchema),
 }
 
-impl Constraint<JsonValue> for ArrayItemConstraints {
+impl Constraint for ArrayItemConstraints {
+    fn combine(
+        self,
+        other: Self,
+    ) -> Result<(Self, Option<Self>), Report<ResolveClosedDataTypeError>> {
+        match (self, other) {
+            (Self::Boolean(lhs), Self::Boolean(rhs)) => lhs
+                .combine(rhs)
+                .map(|(lhs, rhs)| (Self::Boolean(lhs), rhs.map(Self::Boolean))),
+            (Self::Number(lhs), Self::Number(rhs)) => lhs
+                .combine(rhs)
+                .map(|(lhs, rhs)| (Self::Number(lhs), rhs.map(Self::Number))),
+            (Self::String(lhs), Self::String(rhs)) => lhs
+                .combine(rhs)
+                .map(|(lhs, rhs)| (Self::String(lhs), rhs.map(Self::String))),
+            _ => bail!(ResolveClosedDataTypeError::IntersectedDifferentTypes),
```

Comment:
you are using language that's a bit confusing, there is `combine`, `intersect` and `fold`, which one is it?

Reply:
Renamed it to `intersection`